index_node/esg_cog.py

```diff
@@ -112,30 +111,26 @@ def setup_cog(COG_DIR="/usr/local/cog"):
         transfer_api_client_python(os.path.join(COG_DIR, "transfer-api-client-python"))
 
         # setup CoG database and configuration
-        esg_functions.stream_subprocess_output("python setup.py install")
+        esg_functions.call_binary("python", ["setup.py", "install"])
 
         # create or upgrade CoG installation
-        esg_functions.stream_subprocess_output("python setup.py setup_cog --esgf=true")
+        esg_functions.call_binary("python", ["setup.py", "setup_cog", "--esgf=true"])
 
         # collect static files to ./static directory
-        esg_functions.stream_subprocess_output("python manage.py collectstatic --no-input")
+        esg_functions.call_binary("python", ["manage.py", "collectstatic", "--no-input"])
 
     # create non-privileged user to run django
+    esg_functions.add_unix_group("cogadmin")
     try:
-        esg_functions.stream_subprocess_output("groupadd -r cogadmin")
-    except SubprocessError, error:
-        logger.debug(error.__dict__["data"]["returncode"])
-        if error.__dict__["data"]["returncode"] == 9:
-            pass
-    try:
-        esg_functions.stream_subprocess_output("useradd -r -g cogadmin cogadmin")
-    except SubprocessError, error:
-        logger.debug(error.__dict__["data"]["returncode"])
-        if error.__dict__["data"]["returncode"] == 9:
+        esg_functions.call_binary("useradd", ["-r", "-g", "cogadmin", "cogadmin"])
+    except ProcessExecutionError, err:
+        if err.retcode == 9:
             pass
```

Comment:
Same comment from other reveiw, have a `create_user` function?

Reply:
I'll create a separate issue and add the create_user function